### PR TITLE
[PR] Implement Sign-In & Logout (Front End & Back End)

### PR DESCRIPTION
**Pull request by raccoonrick**
**Description:**

This PR adds user sign-in and logout functionality, along with session handling using tokens to manage authentication across the application. The frontend and backend logic for the sign-in page are fully integrated, allowing users to securely log in and out. A token-based session management system is implemented to maintain user authentication throughout their session, enhancing security and user experience. This feature ensures that user sessions are consistently tracked, enabling smooth transitions between authenticated states across the application. 

This PR addresses the requirements outlined in issue #46
This PR resolves issue #65

### CODE REVIEW

File: .env
Comment (MyWaterBottle) [inline .env:1]: Will need to rotate your credentials

**Review by MyWaterBottle — DISMISSED** [review]
.env have credentials

File: .env
Comment (raccoonrick) [reply .env:1]: Just updated the env file thanks

**Comment by raccoonrick** [comment]
Need to fix CI workflow errors

**Comment by harismah** [comment]
I found an issue with logger.js in the backend server folder, due to using hardcoded path (/app/logs/app.log) it wasn't working on my OS, i'm using macOs so I believe paths like /app/logs might not cause an issue on windows but. On macOS (and Linux), the /app directory refers to the root directory (/), which is a restricted system path. Most users do not have permissions to create directories or files in the root of the file system.

**Comment by harismah** [comment]
Additionally, It seems the styling for the home page Button has change a little bit from the main branch
Screenshot 2024-10-12 at 10 47 11 PM


Other than that everything seems to be working appropriately:
- Signed in successfully
- Sign In page redirecting me to homepage after signing in and providing the logout option
- Successfully storing the token in local storage

Screenshot 2024-10-12 at 10 47 25 PM
Screenshot 2024-10-12 at 10 48 29 PM
Screenshot 2024-10-12 at 10 49 48 PM

**Comment by raccoonrick** [comment]
Logger and signin page should be fixed based on the comments

**Review by harismah — APPROVED** [review]
Perfect, everything is working on my end. The points mentioned were fixed. The session token is being stored in local database appropriately and is being removed once logged out.
Screenshot 2024-10-13 at 11 36 21 AM
Screenshot 2024-10-13 at 11 36 31 AM
Screenshot 2024-10-13 at 11 36 38 AM

**Review by harismah — APPROVED** [review]

**Pull request merged**